metaflow/plugins/events_decorator.py

```diff
@@ -188,6 +241,63 @@ def flow_init(
 
         # TODO: Handle scenario for local testing using --trigger.
 
+    def format_deploytime_value(self):
+        for trigger in self.triggers:
+            # Case were trigger is a function that returns a list of events
+            # Need to do this bc we need to iterate over list later
+            if isinstance(trigger, DeployTimeField):
+                if isinstance(deploy_time_eval(trigger), list):
+                    deploy_val = deploy_time_eval(trigger)
+                    self.triggers.remove(trigger)
```

Comment:
it's typically dangerous to modify lists like this while they are being iterated on. I'd recommend not touching self.triggers but forming something like `new_triggers` and updating at the very end.